src/NzbDrone.Core/MediaFiles/EpisodeImport/ImportDecisionMaker.cs

```diff
@@ -138,10 +151,59 @@ private string EvaluateSpec(IRejectWithReason spec, LocalEpisode localEpisode)
                 //e.Data.Add("report", remoteEpisode.Report.ToJson());
                 //e.Data.Add("parsed", remoteEpisode.ParsedEpisodeInfo.ToJson());
                 _logger.ErrorException("Couldn't evaluate decision on " + localEpisode.Path, e);
-                return string.Format("{0}: {1}", spec.GetType().Name, e.Message);
+                return String.Format("{0}: {1}", spec.GetType().Name, e.Message);
             }
 
             return null;
         }
+
+        private bool ParseFolder(List<string> videoFiles, Series series, ParsedEpisodeInfo folderInfo)
+        {
+            var files = new List<string>();
+
+            if (folderInfo == null)
+            {
+                return false;
+            }
+
+            if (folderInfo.FullSeason)
+            {
+                return false;
+            }
+
+            foreach (var file in videoFiles)
+            {
+                var size = _diskProvider.GetFileSize(file);
+                var fileQuality = QualityParser.ParseQuality(file);
+                var parseResult = Parser.Parser.ParsePath(file);
+
+                if (parseResult == null)
+                {
+                    return false;
+                }
+
+                var sample = _sampleService.IsSample(series, GetQuality(folderInfo, fileQuality, series), file, size, parseResult.SeasonNumber);
+
+                if (!sample)
+                {
+                    files.Add(file);
```

Comment:
No, could be multiple files, won't know if they're samples or not until we loop through them all.